plugins/discovery-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java

```diff
@@ -19,31 +19,25 @@
 
 package org.elasticsearch.cloud.azure.management;
 
+import com.microsoft.windowsazure.core.utils.KeyStoreType;
 import com.microsoft.windowsazure.management.compute.models.HostedServiceGetDetailedResponse;
 import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.discovery.azure.AzureUnicastHostsProvider;
 
-import java.util.Locale;
-
-/**
- *
- */
 public interface AzureComputeService {
 
-    static public final class Management {
-        public static final String API_IMPLEMENTATION = "cloud.azure.management.api.impl";
-
-        public static final String SUBSCRIPTION_ID = "cloud.azure.management.subscription.id";
-        public static final String SERVICE_NAME = "cloud.azure.management.cloud.service.name";
+    final class Management {
+        public static final Setting<String> SUBSCRIPTION_ID_SETTING = Setting.simpleString("cloud.azure.management.subscription.id", false, Setting.Scope.CLUSTER);
+        public static final Setting<String> SERVICE_NAME_SETTING = Setting.simpleString("cloud.azure.management.cloud.service.name", false, Setting.Scope.CLUSTER);
 
         // Keystore settings
-        public static final String KEYSTORE_PATH = "cloud.azure.management.keystore.path";
-        public static final String KEYSTORE_PASSWORD = "cloud.azure.management.keystore.password";
-        public static final String KEYSTORE_TYPE = "cloud.azure.management.keystore.type";
+        public static final Setting<String> KEYSTORE_PATH_SETTING = Setting.simpleString("cloud.azure.management.keystore.path", false, Setting.Scope.CLUSTER);
+        public static final Setting<String> KEYSTORE_PASSWORD_SETTING = Setting.simpleString("cloud.azure.management.keystore.password", false, Setting.Scope.CLUSTER);
+        public static final Setting<KeyStoreType> KEYSTORE_TYPE_SETTING = new Setting<>("cloud.azure.management.keystore.type", KeyStoreType.pkcs12.name(), KeyStoreType::fromString, false, Setting.Scope.CLUSTER);
```

Comment:
Thanks @danielmitterdorfer. So I think we should modify https://github.com/elastic/elasticsearch/pull/16291/files#diff-3d1a858d84f619a449e4a1d8829fc69eR91 and fails instead of fallbacking so we are consistent? WDYT?